openslides/users/access_permissions.py

```diff
@@ -185,21 +185,23 @@ def get_restricted_data(self, container, user):
         # Expand full_data to a list if it is not one.
         full_data = container.get_full_data() if isinstance(container, Collection) else [container.get_full_data()]
 
-        # Parse data.
-        for full in full_data:
-            if full['user_id'] == user.id:
-                data = [full]
-                break
-        else:
-            data = []
-
-        # Reduce result to a single item or None if it was not a collection at
-        # the beginning of the method.
-        if isinstance(container, Collection):
-            restricted_data = data
-        elif data:
-            restricted_data = data[0]
-        else:
-            restricted_data = None
+        restricted_data = None
+        if user is not None:
+            # Parse data.
```

Comment:
Please move `full_data = ...` into the condition block. Also add a short comment behind line 188 that anonymous users also get None because they can not have personal notes.